sdk/couchbase/tests/common/mod.rs

```diff
@@ -33,9 +34,15 @@ where
     Fut: std::future::Future<Output = Result<Option<T>, couchbase::error::Error>>,
 {
     while Instant::now() < deadline {
-        if let Ok(Some(r)) = f().await {
+        let res = f().await;
+        if let Ok(Some(r)) = res {
             return r;
         }
+
+        if let Err(e) = res {
+            error!("{:?}", e);
+        }
```

Comment:
Why not just a match?
The sleep line below could also be in the second match clause

Reply:
It just kind of organically evolved, updated.